tools/template_based_query_generation/src/main/java/token/Tokenizer.java

```diff
@@ -200,11 +211,34 @@ private void generateInsertExp(Token token) {
   }
 
   /**
-   * TODO: needs a function to generate a few rows of data to insert into table
+   * TODO: fix with Utils string builder to increase efficiency
+   * sets token to be a values expression
    * @param token
    */
   private void generateValuesExp(Token token) {
-
+    int placeHolder = generateNextPlaceHolder(token.getTokenInfo().getTokenType());
+    int numRows = r.nextInt(this.maxNumColumnsValues) + 1;
+    ArrayList<Column> values = this.table.generateData(numRows);
```

Comment:
Should we still use ArrayList<Column> here?

Reply:
Yes, fixed this. Writing up some manual tests and pushing afterwards